packages/dcos-integration-test/extra/conftest.py

```diff
@@ -30,17 +32,39 @@ def dcos_api_session(dcos_api_session_factory):
     return api
 
 
+def pytest_cmdline_main(config):
+    user_outside_cluster = True
+    if os.path.exists('/opt/mesosphere/bin/dcos-shell'):
```

Comment:
I think, this "feature" should be commented or explained as why we choose to do this. Otherwise the rest of the changes look good to me.